src/OpenTelemetry.Instrumentation.AspNetCore/README.md

```diff
@@ -130,6 +130,23 @@ is the general extensibility point to add additional properties to any activity.
 The `Enrich` option is specific to this instrumentation, and is provided to
 get access to `HttpRequest` and `HttpResponse`.
 
+## How this instrumentation works?
+
+This instrumentation is a [special case of
+instrumentation.](../../../docs/trace/extending-the-sdk/README.md#special-case--instrumentation-for-libraries-producing-legacy-activity)
+The target library (ASP.NET Core), *already* produces an `Activity`, using the
+legacy method and also fires DiagnosticSource event. Hence, the `Activity` will
+have its `Kind` set to internal, and will have its ActivitySource as empty
+string, and will have no tags populated. This instrumentation leverages the
+DiagnosticSource callbacks provided by the ASP.NET Core library, and modifies
+the activity to set its Kind, populate Tags as per Otel semantic conventions
+etc. It is important to note that this enrichment occurs *after* sampler is
+executed. In other words, when Sampler sees this activity, the Kind will be
+internal and have almost no Tags populated. To achieve any advanced filtering of
+the activity based on its Tags, it is hence recommended to use the
+[Filter])(#filter) option provided by this instrumentation.
```

Comment:
@utpilla We'll need to do the following if the Filter decided to drop the activity.

```
activity.IsAllDataRequested = false;
activity.ActivityTraceFlags &= ~ActivityTraceFlags.Recorded;
```